homeassistant/components/switch/wemo.py

```diff
@@ -141,12 +153,22 @@ def is_on(self):
     @property
     def available(self):
         """True if switch is available."""
-        if self.wemo.model_name == 'Insight' and self.insight_params is None:
+        if self._model_name == 'Insight' and self.insight_params is None:
+            return False
+        if self._model_name == 'Maker' and self.maker_params is None:
             return False
-        if self.wemo.model_name == 'Maker' and self.maker_params is None:
+        if self._model_name == 'CoffeeMaker' and self.coffeemaker_mode is None:
             return False
         return True
 
+    @property
+    def icon(self):
+        """Icon of device based on its type."""
+        if self.coffeemaker_mode is not None:
+            return 'mdi:coffee'
+        else:
+            return SwitchDevice.icon
```

Comment:
@robbiet480 Ah, but there's more validity to this device than you're giving it credit for. The coffee maker reports every status change--e.g. the carafe being removed and re-seated... this could be used as a pretty reliable "user is definitely awake" trigger.